scripts/mbedtls_framework/code_wrapper/psa_wrapper.py

```diff
@@ -0,0 +1,257 @@
+            if self._parameter_should_be_copied(function.name,
+                                                function.arguments[param.index].name))
+
+        BufferParameter.poison_multi_write(out, buffer_parameters, True)
+        super()._write_function_call(out, function, argument_names)
+        BufferParameter.poison_multi_write(out, buffer_parameters, False)
+
+    def _skip_function(self, function: c_wrapper_generator.FunctionInfo) -> bool:
+        if function.return_type != 'psa_status_t':
+            return True
+        if function.name in self._SKIP_FUNCTIONS:
+            return True
+        return False
+
+    def _return_variable_name(self,
+                              function: c_wrapper_generator.FunctionInfo) -> str:
+        """The name of the variable that will contain the return value."""
+
+        if function.return_type == 'psa_status_t':
+            return 'status'
+        return super()._return_variable_name(function)
+
+    def _write_prologue(self, out: typing_util.Writable, header: bool) -> None:
+        super()._write_prologue(out, header)
+
+        prologue = []
+        if self._CPP_GUARDS:
+            prologue.append("#if {}".format(self._CPP_GUARDS))
+            prologue.append('')
+
+        for include in self._PSA_WRAPPER_INCLUDES:
+            prologue.append("#include {}".format(include))
+
+        # Make certain there is an empty line at the end of this section.
+        for i in [-1, -2]:
+         if prologue[i] != '':
+            prologue.append('')
+
+        out.write("\n".join(prologue))
+
+    def _write_epilogue(self, out: typing_util.Writable, header: bool) -> None:
+        if self._CPP_GUARDS:
+            out.write("#endif /* {} */\n\n".format(self._CPP_GUARDS))
+        super()._write_epilogue(out, header)
+
+class PSALoggingWrapper(PSAWrapper, c_wrapper_generator.Logging):
+    """Generate a C source file containing wrapper functions that log PSA Crypto API calls."""
+
+    def __init__(self,
+                 stream: str,
+                 output_h_f: str,
+                 output_c_f: str,
+                 in_headers:  Collection[str] = DEFAULTS["input_headers"],
+                 config: Dict[str, Any]= DEFAULTS) -> None:
+
+        super().__init__(output_h_f, output_c_f, in_headers, config)
+        self.set_stream(stream)
+
+    _PRINTF_TYPE_CAST = c_wrapper_generator.Logging._PRINTF_TYPE_CAST.copy()
+    _PRINTF_TYPE_CAST.update({
+        'mbedtls_svc_key_id_t': 'unsigned',
+        'psa_algorithm_t': 'unsigned',
+        'psa_drv_slot_number_t': 'unsigned long long',
+        'psa_key_derivation_step_t': 'int',
+        'psa_key_id_t': 'unsigned',
+        'psa_key_slot_number_t': 'unsigned long long',
+        'psa_key_lifetime_t': 'unsigned',
+        'psa_key_type_t': 'unsigned',
+        'psa_key_usage_flags_t': 'unsigned',
+        'psa_pake_role_t': 'int',
+        'psa_pake_step_t': 'int',
+        'psa_status_t': 'int'
```

Comment:
```suggestion
        'psa_status_t': 'int',
```

https://stackoverflow.com/questions/11597901/why-are-trailing-commas-allowed-in-a-list